instana/instrumentation/boto3_inst.py

```diff
@@ -35,6 +46,20 @@ def lambda_inject_context(payload, scope):
             logger.debug("non-fatal lambda_inject_context: ", exc_info=True)
 
 
+    @wrapt.patch_function_wrapper("botocore.hooks", "HierarchicalEmitter.emit")
```

Comment:
This hook gets called around a dozen of times per each request, which might have some performance impacts.
Regardless of the performance aspect, the main issue is that this simply doesn't always work.
Since there is no offical way of adding headers, the customer might get the idea [from here](https://stackoverflow.com/a/66097613), to add the headers with with `event_system.register_first('before-sign.s3.CreateBucket', _add_header)`, and in that case unfortunately this hook is too early and doesn't seem to work.

Reply:
Made changes to extract headers from `add_auth` method of `botocore.auth`. This way, the hook gets called only once per api_call and adapts to both `before-call` and `before-sign` events.